.github/workflows/validation.yaml

```diff
@@ -66,3 +70,125 @@ jobs:
+      - name: Remove format labels
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { owner, repo } = context.repo;
+            const prNumber = context.payload.pull_request.number;
+            
+            // Remove the label if it exists
+            const existingLabels = await github.rest.issues.listLabelsOnIssue({
+              owner,
+              repo,
+              issue_number: prNumber,
+            });
+            const labelsToRemove = existingLabels.data.map(label => label.name).filter( label => label.startsWith("format:") );
+            for (const labelToRemove of labelsToRemove) {
+              await github.rest.issues.removeLabel({
+                owner,
+                repo,
+                issue_number: prNumber,
+                name: labelToRemove,
+              });
+            }
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: 'temurin'
+          cache: maven
+      - name: Check
+        id: spotlessCheck
+        run: |
+          set -x -e -o pipefail
+          mvn -V -e -B -ntp spotless:check
+      - name: Prepare failure message
+        if: ${{ failure() && steps.spotlessCheck.conclusion == 'failure' }}
+        run: |
+          set -x -e -o pipefail
+          mvn -V -e -B -ntp spotless:apply
+          git diff --output=/tmp/full-diff.txt 
+          git diff --name-only --output=/tmp/unformatted.txt
+          _files=$(cat /tmp/unformatted.txt | wc -l)
+          echo "### Format Checker Report" >> /tmp/message
+          echo "" >> /tmp/message
+          echo "There are **${_files} files** with format errors." >> /tmp/message
+          echo "" >> /tmp/message
+          echo '* To fix the build, please run `mvn spotless:apply` in your branch and commit the changes.' >> /tmp/message 
+          echo '* Alternatively, apply the `auto-format` label to the PR.' >> /tmp/message
+          echo "" >> /tmp/message
+          echo "" >> /tmp/message
+          echo "<details>" >> /tmp/message
+          echo "  <summary>Files with format issues in this PR</summary>" >> /tmp/message
+          echo "" >> /tmp/message
+          echo '```' >> /tmp/message
+          cat /tmp/unformatted.txt  >> /tmp/message
+          echo '```' >> /tmp/message
+          echo "</details>" >> /tmp/message
+          echo "" >> /tmp/message
+          echo "" >> /tmp/message
+          echo "<details>" >> /tmp/message
+          echo "  <summary>Full Diff</summary>" >> /tmp/message
+          echo "" >> /tmp/message
+          echo '```patch' >> /tmp/message
+          cat /tmp/full-diff.txt  >> /tmp/message
+          echo '```' >> /tmp/message
+          echo "</details>" >> /tmp/message
```

Comment:
Maybe use some template and do some replace operation on it?

Reply:
Good point.
I'll not have much time to fix it next week. 
Perhaps it can be changed later on?